integrationTests/multiShard/smartContract/scCallingSC_test.go

```diff
@@ -419,6 +423,124 @@ func TestSCCallingInCrossShard(t *testing.T) {
 	}
 }
 
+func TestSCCallingDNSUserNames(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	numOfShards := 2
+	nodesPerShard := 2
+	numMetachainNodes := 2
+
+	advertiser := integrationTests.CreateMessengerWithKadDht("")
+	_ = advertiser.Bootstrap()
+
+	genesisFile := "testdata/smartcontracts.json"
+	nodes := integrationTests.CreateNodesWithFullGenesis(
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		integrationTests.GetConnectableAddress(advertiser),
+		genesisFile,
+	)
+
+	idxProposers := make([]int, numOfShards+1)
```

Comment:
Optional / create this map perhaps using explicit declaration instead? For example:

```
var idxProposers = []int{ 0, 2, 4 }
```
and move it below the

```
numOfShards := 2
nodesPerShard := 2
numMetachainNodes := 2
```

I know the test would be less dynamic in terms of adjusting its parameters this way - but it would be a bit easier to read what's important within the test and ignore some setup that is somehow of a secondary concern.

Reply:
it is the same way of creating basic test setup for almost all integration tests.